src/libstd/sys/redox/ext/process.rs

```diff
@@ -45,12 +45,33 @@ pub trait CommandExt {
     /// like `malloc` or acquiring a mutex are not guaranteed to work (due to
     /// other threads perhaps still running when the `fork` was run).
     ///
+    /// This also means that all resources such as file descriptors and
+    /// memory-mapped regions got duplicated. It is your responsibility to make
+    /// sure that the closure does not violate library invariants by making
+    /// invalid use of these duplicates.  Moreover, POSIX demands that you only
+    /// perform operations that are explicitly documented as async-signal-safe.
```

Comment:
Er sorry this is something I hadn't researched at all so I was hoping to expand this a bit more with some research rather than include it literally. Checking [these docs](http://pubs.opengroup.org/onlinepubs/009695399/functions/fork.html) I only find a few mentions of async signal safe:

> Consequently, to avoid errors, the child process may only execute async-signal-safe operations until such time as one of the exec functions is called

followed by 

> When the application calls fork() from a signal handler and any of the fork handlers registered by pthread_atfork() calls a function that is not asynch-signal-safe, the behavior is undefined.

That's all I can find which leads me to think https://github.com/rust-lang/rust/issues/39575#issuecomment-277789018 may be not-quite-correct claim. It sounds like it's practically UB to `fork` in a signal handler, but otherwise it just says 'to avoid errors', not that it's UB to execute non-async-signal-safe code after a fork.

If that's all the case then I don't think posix should be mentioned after all, what do you think?

Reply:
I am not at all a POSIX expert, but your analysis makes sense to me.